docs/l10n.rst

```diff
@@ -9,17 +9,421 @@ Localization
+locale. English locales (e.g. en-US, en-GB) will never use the fallback and will print the simplified
+english version of the primary string if not overridden in the more specific locale.
+
+You can also pass in replacement variables into the ``ftl()`` function for use with `fluent variables`_.
+If you had a variable in your fluent file like this:
+
+.. code-block:: fluent
+
+    welcome = Welcome, { $user }!
+
+You could use that in a template like this:
+
+.. code-block:: jinja
+
+    <h2>{{ ftl('welcome', user='Dude') }}<h2>
+
+For our purposes these are mostly useful for things that can change, but which shouldn't involve
+retranslation of a string (e.g. URLs or email addresses).
+
+This helper is available in Jinja templates and Python code in views. For use in a view you should
+always call it in the view itself:
+
+.. code-block:: python
+
+    # views.py
+    from lib.l10n_utils import render
+    from lib.l10n_utils.fluent import ftl
+
+    def about_view(request):
+        hello_string = ftl('about-hello')
+        render(request, 'about.html', {'hello': hello_string}, ftl_files='about')
+
+If you need to use this string in a view, but define it outside of the view itself, you can use the
+``ftl_lazy`` variant which will delay evaluation until render time. This is mostly useful for defining
+messages shared among several views in constants in a ``views.py`` or ``models.py`` file.
+
+The ``ftl_has_messages`` helper function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Another useful template too is the ``ftl_has_messages()`` function. You pass it any number
+of string IDs and it will return ``True`` only if all of those message IDs exist in the current
+translation. This is useful when you want to add a new block of HTML to a page that is already
+translated, but don't want it to appear untranslated on any page.
+
+.. code-block:: jinja
+
+    {% if ftl_has_messages('new-title', 'new-description') %}
+      <h3>{{ ftl('new-title') }}</h3>
+      <p>{{ ftl('new-description') }}</p>
+    {% else %}
+      <h3>{{ ftl('title') }}</h3>
+      <p>{{ ftl('description') }}</p>
+    {% endif %}
+
+If you'd like to have it return true when any of the given message IDs exist in the translation
+instead of requiring all of them, you can pass the optional ``require_all=False`` parameter and
+it will do just that.
+
+There is a version of this function for use in views called ``has_messages``. It works exactly the
+same way but is meant to be used in the view Python code.
+
+.. code-block:: python
+
+    # views.py
+    from lib.l10n_utils import render
+    from lib.l10n_utils.fluent import ftl, has_messages
+
+    def about_view(request):
+        if has_messages('about-hello-v2', 'about-title-v2'):
+            hello_string = ftl('about-hello-v2')
+            title_string = ftl('about-title-v2')
+        else:
+            hello_string = ftl('about-hello')
+            title_string = ftl('about-title')
```

Comment:
Can we turn this example into a feature switch thing? Like, 

```python
    use_skyline_layout = has_messages('about-hello-v2', 'about-title-v2')
    render(request, 'about.html', {'use_skyline_layout': use_skyline_layout}, ftl_files='about')
```

You can probably come up with a better flag name. I hope that that's closer what will happen in practice?